### PR TITLE
Issue/#30

### PR DESCRIPTION
**Pull request by HannesSundin**
refactored login page and the used components.
some views moved to temp/folder, to be used in staff registration
and some views moved into index.js

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -23,15 +24,31 @@ export default function Index() {
 			<main className={styles.main}>
```
Comment (StelFoog) [inline pages/index.js:24]: Everything in main should be it's own non-reusable component (in this same file) to match the MVP-pattern, e.g.
```js
function LoginForm({pnr, setPnr, password, setPassword, handleSubmit}) {
  return (
    <form
      onSubmit={e => {
        e.preventDefault();
        handleSubmit();
      }}
    >
      {/* TODO */}
    </form>
}


export default Index() {
  const [pnr, setPnr] = useState("");
  const [password, setPassword] = useSate("");
  
  function handleSubmit() {
    /* todo */
  }

  return (
    <>
      <Head>{/*...*/}</Head>
      <main>
        <LoginForm pnr={pnr} setPnr={setPnr} password={password} setPassword={setPassword} handleSubmit={handleSubmit}  />
        {/* Other things */}
      </main>
    </>
  )
}
```

File: components/ErrorTextView.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import styles from "../styles/Home.module.css";
+
+export default function ErrorTextView(props) {
+    return (props.isDisabled || <div className={styles.error}>
+        {props.error}
+    </div>)
+}
```
Comment (StelFoog) [inline components/ErrorTextView.js:7]: It's a component and should be named as such, file should probably be called `ErrorText.js`

File: components/RequiredTextView.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import styles from "../styles/Home.module.css";
+
+export default function RequiredTextview() {
+    return (<span className={styles.required}> *required {' '} </span>)
+}
```
Comment (StelFoog) [inline components/RequiredTextView.js:5]: It's a component and should be named as such, file should probably be called `ErrorText.js`

Comment (StelFoog) [inline components/RequiredTextView.js:5]: missing EOF

**Review by StelFoog — CHANGES_REQUESTED** [review]

File: pages/index.js [hunk 4]
```diff
@@ -23,15 +24,31 @@ export default function Index() {
...
+							Password {' '}
+							<input id="password" placeholder="Password" type="password" required
+								   onChange={(change) => {
+									  setPassword(change.target.value)
+								   }}/>
+							{!password && <RequiredTextView/>}
+						</label>
+						<br/>
+					</form>
+					<button
```
[the remark this answers is not in the document]
Comment (HannesSundin) [reply pages/index.js]: commit e65371554d00de7a0aea883321e5c7829446b76a

File: pages/index.js [hunk 5]
```diff
@@ -23,15 +24,31 @@ export default function Index() {
...
+						disabled={!pnr || !password && pnr.length !== 10}
+						onClick={() => {
```
[the remark this answers is not in the document]
Comment (HannesSundin) [reply pages/index.js]: commit e65371554d00de7a0aea883321e5c7829446b76a

File: pages/index.js (hunk 1 above)
Comment (HannesSundin) [reply pages/index.js]: commit e65371554d00de7a0aea883321e5c7829446b76a

File: components/ErrorTextView.js (hunk 2 above)
Comment (HannesSundin) [reply components/ErrorTextView.js]: commit e65371554d00de7a0aea883321e5c7829446b76a

File: components/RequiredTextView.js (hunk 3 above)
Comment (HannesSundin) [reply components/RequiredTextView.js]: commit e65371554d00de7a0aea883321e5c7829446b76a

Comment (HannesSundin) [reply components/RequiredTextView.js]: commit e65371554d00de7a0aea883321e5c7829446b76a

File: pages/index.js [hunk 6]
```diff
@@ -1,64 +1,91 @@
...
-	const [pnr, setPnr] = useState('');
-	const [error, setError] = useState('');
+    const [password, setPassword] = useState('');
+    const [pnr, setPnr] = useState('');
+    const [error, setError] = useState('');
+    const router = useRouter();
+    const dispatch = useDispatch();
+
+    function handleSubmit() {
+        fetch('/api/staff/login', {
```
[the remark this answers is not in the document]
Comment (HannesSundin) [reply pages/index.js:50]: do you mean to check if PIN contains numbers?
pressing enter while the button is disabled does nothing for me.

[the remark this answers is not in the document]
Comment (StelFoog) [reply pages/index.js:50]: strange seems to have fixed itself

**Pull request merged**